pkg/controller/hibernation/hibernation_controller_test.go

```diff
@@ -1270,3 +1272,138 @@ func Test_timeBeforeClusterSyncCheck(t *testing.T) {
 		})
 	}
 }
+
+func Test_cumulativeMetric_value(t *testing.T) {
+	logger := log.New()
+	logger.SetLevel(log.DebugLevel)
+
+	scheme := runtime.NewScheme()
+	corev1.AddToScheme(scheme)
+	batchv1.AddToScheme(scheme)
+	configv1.AddToScheme(scheme)
+	hivev1.AddToScheme(scheme)
+	hiveintv1alpha1.AddToScheme(scheme)
+	machineapi.AddToScheme(scheme)
+
+	cdBuilder := testcd.FullBuilder(namespace, cdName, scheme).Options(
+		testcd.Installed(),
+		testcd.WithClusterVersion("4.4.9"),
+	)
+	o := clusterDeploymentOptions{}
+	csBuilder := testcs.FullBuilder(namespace, cdName, scheme).Options(
+		testcs.WithFirstSuccessTime(time.Now().Add(-10 * time.Hour)),
+	)
+
+	tests := []struct {
+		name               string
+		cd                 *hivev1.ClusterDeployment
+		cs                 *hiveintv1alpha1.ClusterSync
+		setupActuator      func(actuator *mock.MockHibernationActuator, cd *hivev1.ClusterDeployment)
+		setupRemote        func(builder *remoteclientmock.MockBuilder)
+		setupMetricLogging func(observerFunc *prometheus.Observer)
+		//metric             *metrics.HiveMetric
+		labels    string
+		timeValue float64
+	}{
+		{
+			name: "stopping, machines have stopped",
+			cd: cdBuilder.Options(o.shouldHibernate, o.stopping,
+				testcd.WithCondition(hivev1.ClusterDeploymentCondition{
+					Type:               hivev1.ClusterReadyCondition,
+					Status:             corev1.ConditionFalse,
+					Reason:             hivev1.ReadyReasonStoppingOrHibernating,
+					LastTransitionTime: metav1.Time{Time: time.Now().Add(-2 * time.Hour)},
+				})).Build(),
+			cs: csBuilder.Build(),
+			setupActuator: func(actuator *mock.MockHibernationActuator, cd *hivev1.ClusterDeployment) {
+				// Simulate changing the order of cluster deployment conditions mid reconcile
+				controllerutils.SetClusterDeploymentCondition(cd.Status.Conditions,
```

Comment:
When I debugged with this changes, setting the condition here didn't work - this condition wasn't present at all on the CD. Looking for a better way to enforce it